### DISCUSSION TITLE
Syslog logging via Quarkus properties

### DISCUSSION THREAD

**Discussion by dasniko** [keycloak/keycloak#10201] [Archive]
Hi @DGuhr 

in one of my customer-projects, we are using Syslog logging with Keycloak.
I got it working with native Quarkus properties in `conf/quarkus.properties` like suggested in the docs. Works.

Unfortunately, I cannot overwrite any of the Quarkus properties at runtime, as this seams to be a build time config. But there is at least the `quarkus.log.syslog.endpoint` property, which I want to set dynamically in the various environments to different values, or even switch it off with `quarkus.log.syslog.enable=false`. While I'm in a Docker container, I try to set the overwrites via environment variables (uppercase letters and `_` instead of `.`.
But this doesn't work, only the build-time properies get into count.

Do you have any suggestions, workarounds or even fixes for upcoming versions?

**Comment by DGuhr** [comment]
Hey. To be honest, the current logging impl is more a "better than nothing" v1. I'll start to add more capabilities shortly. I'd focus on json support, and was not sure about syslog, but seems we should also support it. So yes, expect more to come in upcoming versions :) We always try to give the most convenient ux, so would be nice to know what raw quarkus properties you use in the syslog area, and how we could make your life easier, e.g. if some properties could have a good abstraction (we are able to provide e.g. bwtter naming, 1:n mappings)...). I am more than open to input, as I dont have much experience using syslog.

**Reply by dasniko** [reply]
Thanks for your reply.

Currently, I'm using this set of properties:

```
quarkus.log.syslog.enable=true
quarkus.log.syslog.endpoint=hostname:port
quarkus.log.syslog.protocol=ssl-tcp
quarkus.log.syslog.app-name=keycloak
quarkus.log.syslog.format=[%c{3.}] (%t) %s%e%n
quarkus.log.syslog.truncate=false
```

So, at least in my usecase I need to set the `quarkus.log.syslog.enable` and `quarkus.log.syslog.endpoint` properties dynamically, depending on the environment I'm running in (e.g. dev, test, prod, etc.).

(Of course, there are more syslog properties at the [Quarkus configuration page](https://quarkus.io/guides/all-config) than the ones I used above, it might be good to support more than only the above mentioned.)

**Comment by dasniko** [comment]
In https://github.com/keycloak/keycloak/discussions/9592#discussioncomment-2033394 @stianst mentioned the use of `quarkus.properties` as an extension rather than a configuration.

Yes, I kind of _extend_ Keycloak with the syslog logging capabilities of Quarkus, but I also need to _configure_ this for various envs with specific values.

**Reply by codespearhead** [reply]
Should this be closed now that #10414 was closed too?

**Comment by dasniko** [comment]
Any news on this, @DGuhr ?

**Comment by DGuhr** [comment]
@dasniko atm we think about letting people use quarkus.properties directly instead, meaning without the gate of "everything is a build time property" to support edge cases like this. I guess that would solve your problem?

**Reply by dasniko** [reply]
Yeah, great news. This would indeed solve my issue! 👍
(and also allow further direct quarkus config, if needed)
Thank you! 🙏

**Reply by dasniko** [reply]
One more question... Will it only be possible via `quarkus.properties` file of also via Quarkus environment variables?

**Comment by cgeorgilakis** [comment]
@DGuhr any progress on this? 
We have the same problem using version 18. 
See [the new discussion](https://github.com/keycloak/keycloak/discussions/17691) I have created before showing this old github discussion.
